### PR TITLE
MAKE-534: methods for FTDC chunks to calculate rollups/means/totals from the chunks

### PR DESCRIPTION
**Pull request by julianedwards**

### CODE REVIEW

File: perf/ftdc_rollups.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+			case "Timers.Total":
+				perfStats.timers.total = time.Duration(metric.Values[len(metric.Values)-1])
+			case "Gauges.State":
+				perfStats.gauges.stateTotal += sum(metric.Values)
+			case "Gauges.Workers":
+				perfStats.gauges.workersTotal += sum(metric.Values)
+			case "Gauges.Failed":
+				perfStats.gauges.failedTotal += sum(metric.Values)
+			default:
+				return performanceStatistics{}, errors.New(fmt.Sprintf("unknown field name %s", name))
```
Comment (tychoish) [inline perf/ftdc_rollups.go]: errors.Errorf

File: perf/ftdc_rollups.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+			Value:         float64(s.gauges.stateTotal) / float64(s.numSamples),
+			Version:       defaultVer,
+			UserSubmitted: false,
+		})
+		rollups = append(rollups, model.PerfRollupValue{
+			Name:          "avgWorkers",
+			Value:         float64(s.gauges.workersTotal) / float64(s.numSamples),
+			Version:       defaultVer,
+			UserSubmitted: false,
+		})
```
Comment (tychoish) [inline perf/ftdc_rollups.go]: append is a variadic function so you can do this in one call.

File: perf/ftdc_rollups.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+		Value:         s.counters.size,
+		Version:       defaultVer,
+		UserSubmitted: false,
+	})
+	rollups = append(rollups, model.PerfRollupValue{
+		Name:          "totalSamples",
+		Value:         s.numSamples,
+		Version:       defaultVer,
+		UserSubmitted: false,
+	})
```
Comment (tychoish) [inline perf/ftdc_rollups.go]: merge.

File: perf/ftdc_rollups.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	return rollups
+}
+
+func sum(nums []int64) int64 {
+	sum := int64(0)
+	for _, num := range nums {
+		sum += num
+	}
+	return sum
```
Comment (tychoish) [inline perf/ftdc_rollups.go]: can we just move this to util or use something from stats.

File: perf/ftdc_rollups.go [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+	"github.com/evergreen-ci/sink/model"
+	"github.com/mongodb/ftdc"
+	"github.com/pkg/errors"
+)
+
+const (
+	defaultVer = 1
+)
+
+type performanceStatistics struct {
```
Comment (tychoish) [inline perf/ftdc_rollups.go]: can we merge this with the one in model?

Comment (tychoish) [reply perf/ftdc_rollups.go]: or maybe just use events?

Comment (julianedwards) [reply perf/ftdc_rollups.go]: Once events.Performance has duration as cumulative we can just use that.

Comment (tychoish) [reply perf/ftdc_rollups.go]: let's open a ticket and do this in the next sprint.

**Pull request merged**